### DISCUSSION TITLE
Better recognition

### DISCUSSION THREAD

**Discussion by Lex-Kent** [speechly/speechly#254] [General]
Hi all,

I'm building a real-time character that can be embedded in a web page and chats with the user in natural language.

In my early versions of the app, I used `WebKitSpeechRecognition` where it was supported and the `Google Cloud API` in browsers where it wasn't. I'm now using `Speechly` across all devices because it is so much simpler to use and I can have a single set of functions for all browsers.

However, I am finding that `Speechly's` ability to actually understand the user (or me!) is not as accurate as `WebKitSpeechRecognition` or the Google Cloud API. I assume this is in part because I am English and my tool is built for British English users rather than American English speakers. In `WebKitSpeechRecognition` and the `Google API`, I can specify `en-UK` as the language but I can't seem to do that with `Speechly`.

So I guess I need to train `Speechly`? Which I'm happy to do, but in my case, my character can chat with the user about a huge range of topics.

In one part of my app, my character offers the user 4 choices and asks which the user most associates with - "A", "B", "C" or "D". `Speechly` was hearing all manner of responses like "eh", "hay", "bee", "be", "sea", "say", "si", "day" etc., so as an experiment, I added some training data. And lo and behold, `Speechly` returned the correct response to that question every time. However, the recognition accuracy in the rest of the tool suffered enormously. And when my character asked for the user's name, if they replied "Dee", then of course, the response came back as "D".

I'm also using `OpenAI` in parts of my tool. With `OpenAI` I can steer the response using a particular "prompt" and I can train the AI model using a bunch of different prompts. I think I need the same kind of thing from `Speechly`: When the user is asked their name, I want to target a specific set of training data, when asked the "A", "B", "C" or "D" question, I want to target another set. When asked at what date they want to retire, I want to target a future date entity and so on. And there a _lot_ of questions. When the user is simply chatting with my character using `OpenAI`, then I want to fallback to no training data and to simply rely on `Speechly's` default recognition since I have no idea what the user might say.

Is any of this possible with a single app id?

Many thanks for any help, and apologies for writing such an essay!

**Comment by teelisyys** [comment]
Hi @Lex-Kent 

Thank you for your questions!

Unfortunately we do not currently offer a dedicated model for British English. However, we have found that our acoustic adaptation, which is available on our Enterprise tier, can greatly improve the accuracy by learning domain-specific vocabulary and tuning to acoustic environments, including accents.

Our generally available solution for the different context is to create and train different app ids for them and change between them as needed. However, for our Enterprise tier clients, we have also been able to provide even utterance level context priming.

You may learn more about [our tiers on our website](https://www.speechly.com/pricing) or by [contacting us](https://www.speechly.com/contact).

**Reply by Lex-Kent** [reply]
@teelisyys thanks again for your quick response, it's much appreciated.

Unfortunately $3,000 a month for the enterprise tier is too risky for my client for something that might offer no improvement.
